core/arch/x86_64/kernel/thread_64.S

```diff
@@ -36,6 +32,33 @@ FUNC thread_resume , :
     pushq %r15
     pushq %rdi
     pushq %rsi
+.endm
+
+.macro RESTORE_REGS
+    popq %rsi
+    popq %rdi
+    popq %r15
+    popq %r14
+    popq %r13
+    popq %r12
+    popq %r11
+    popq %r10
+    popq %r9
+    popq %r8
+    popq %rbp
+    popq %rdx
+    popq %rcx
+    popq %rbx
+    popq %rax
+    popfq
+.endm
+
+/* void thread_resume(struct thread_ctx_regs *regs, struct thread_smc_args *args,
+ *                      vaddr_t tmp_stack)
+ */
+FUNC thread_resume , :
+    SAVE_REGS
+    pushq %rsi
```

Comment:
Will this cause confusion? because you push RSI twice. Can we exclude RSI and RDI from SAVE_REGS and RESTORE_REGS?